builtin/logical/pki/cert_util.go

```diff
@@ -330,8 +331,8 @@ func validateNames(req *logical.Request, names []string, role *roleEntry) string
 					}
 				}
 
-				if strings.HasSuffix(sanitizedName, "."+req.DisplayName) ||
```

Comment:
I think this use case is also not needed. It's difficult to reason about a use-case where you would want to make a token display name a globbing CN.